core/src/packet_deserializer.rs

```diff
@@ -130,10 +144,34 @@ impl PacketDeserializer {
     fn deserialize_packets<'a>(
         packet_batch: &'a PacketBatch,
         packet_indexes: &'a [usize],
+        tx_account_lock_limit: usize,
     ) -> impl Iterator<Item = ImmutableDeserializedPacket> + 'a {
-        packet_indexes.iter().filter_map(move |packet_index| {
-            ImmutableDeserializedPacket::new(packet_batch[*packet_index].clone(), None).ok()
-        })
+        packet_indexes
+            .iter()
+            .filter_map(move |packet_index| {
+                ImmutableDeserializedPacket::new(packet_batch[*packet_index].clone(), None).ok()
+            })
+            .filter(move |packet| Self::check_account_locks_limit(packet, tx_account_lock_limit))
+    }
+
+    fn check_account_locks_limit(
+        packet: &ImmutableDeserializedPacket,
+        tx_account_lock_limit: usize,
+    ) -> bool {
+        let message = &packet.transaction().get_message().message;
+        let num_static_accounts = message.static_account_keys().len();
+        let num_looked_up_accounts: usize = message
+            .address_table_lookups()
+            .iter()
+            .map(|address_table_lookup| {
+                address_table_lookup
+                    .iter()
+                    .map(|lookup| lookup.readonly_indexes.len() + lookup.writable_indexes.len())
```

Comment:
TODO: do I add 1 for the lookup table address itself?

Reply:
Looking at `load_lookup_table_addresses` in accounts.rs:
```
    pub fn load_lookup_table_addresses(
        &self,
        ancestors: &Ancestors,
        address_table_lookup: &MessageAddressTableLookup,
        slot_hashes: &SlotHashes,
    ) -> std::result::Result<LoadedAddresses, AddressLookupError> {
        let table_account = self
            .accounts_db
            .load_with_fixed_root(ancestors, &address_table_lookup.account_key)
            .map(|(account, _rent)| account)
            .ok_or(AddressLookupError::LookupTableAccountNotFound)?;

        if table_account.owner() == &solana_address_lookup_table_program::id() {
            let current_slot = ancestors.max_slot();
            let lookup_table = AddressLookupTable::deserialize(table_account.data())
                .map_err(|_ix_err| AddressLookupError::InvalidAccountData)?;

            Ok(LoadedAddresses {
                writable: lookup_table.lookup(
                    current_slot,
                    &address_table_lookup.writable_indexes,
                    slot_hashes,
                )?,
                readonly: lookup_table.lookup(
                    current_slot,
                    &address_table_lookup.readonly_indexes,
                    slot_hashes,
                )?,
            })
        } else {
            Err(AddressLookupError::InvalidAccountOwner)
        }
    }
```
the lookup table is not included in the LoadedAddresses, and therefore will not be included in the lock count.